[matplotplusplus] can not run the simplest official sample

**Describe the bug**
A clear and concise description of what the bug is.

## Env
OS: Windows
matplotplusplus: 2021-04-11#2
Compiler: msvc

## Problem
```C++
#include <cmath>
#include <matplot/matplot.h>

int main() {
    using namespace matplot;
    std::vector<double> x = linspace(0, 2 * pi);
    std::vector<double> y = transform(x, [](auto x) { return sin(x); });

    plot(x, y, "-o");
    hold(on);
    plot(x, transform(y, [](auto y) { return -y; }), "--xr");
    plot(x, transform(x, [](auto x) { return x / pi - 1.; }), "-:gs");
    plot({ 1.0, 0.7, 0.4, 0.0, -0.4, -0.7, -1 }, "k");

    show();
    return 0;
}
```

## Errors
```console
1>matplot.lib(network.cpp.obj) : error LNK2019: unresolved external symbol "class std::vector<struct nodesoup::Point2D,class std::allocator<struct nodesoup::Point2D> > __cdecl nodesoup::fruchterman_reingold(class std::vector<class std::vector<unsigned __int64,class std::allocator<unsigned __int64> >,class std::allocator<class std::vector<unsigned __int64,class std::allocator<unsigned __int64> > > > const &,unsigned int,unsigned int,unsigned int,double,class std::function<void __cdecl(class std::vector<struct nodesoup::Point2D,class std::allocator<struct nodesoup::Point2D> > const &,int)>)" (?fruchterman_reingold@nodesoup@@YA?AV?$vector@UPoint2D@nodesoup@@V?$allocator@UPoint2D@nodesoup@@@std@@@std@@AEBV?$vector@V?$vector@_KV?$allocator@_K@std@@@std@@V?$allocator@V?$vector@_KV?$allocator@_K@std@@@std@@@2@@3@IIINV?$function@$$A6AXAEBV?$vector@UPoint2D@nodesoup@@V?$allocator@UPoint2D@nodesoup@@@std@@@std@@H@Z@3@@Z) referenced in function "protected: void __cdecl matplot::network::process_force_layout(void)" (?process_force_layout@network@matplot@@IEAAXXZ)
1>matplot.lib(network.cpp.obj) : error LNK2019: unresolved external symbol "class std::vector<struct nodesoup::Point2D,class std::allocator<struct nodesoup::Point2D> > __cdecl nodesoup::kamada_kawai(class std::vector<class std::vector<unsigned __int64,class std::allocator<unsigned __int64> >,class std::allocator<class std::vector<unsigned __int64,class std::allocator<unsigned __int64> > > > const &,unsigned int,unsigned int,double,double)" (?kamada_kawai@nodesoup@@YA?AV?$vector@UPoint2D@nodesoup@@V?$allocator@UPoint2D@nodesoup@@@std@@@std@@AEBV?$vector@V?$vector@_KV?$allocator@_K@std@@@std@@V?$allocator@V?$vector@_KV?$allocator@_K@std@@@std@@@2@@3@IINN@Z) referenced in function "protected: void __cdecl matplot::network::process_kawai_layout(void)" (?process_kawai_layout@network@matplot@@IEAAXXZ)
1>matplot.lib(network.cpp.obj) : error LNK2019: unresolved external symbol "class std::vector<double,class std::allocator<double> > __cdecl nodesoup::size_radiuses(class std::vector<class std::vector<unsigned __int64,class std::allocator<unsigned __int64> >,class std::allocator<class std::vector<unsigned __int64,class std::allocator<unsigned __int64> > > > const &,double,double)" (?size_radiuses@nodesoup@@YA?AV?$vector@NV?$allocator@N@std@@@std@@AEBV?$vector@V?$vector@_KV?$allocator@_K@std@@@std@@V?$allocator@V?$vector@_KV?$allocator@_K@std@@@std@@@2@@3@NN@Z) referenced in function "protected: void __cdecl matplot::network::process_force_layout(void)" (?process_force_layout@network@matplot@@IEAAXXZ)
1>\path\to\test_matplotplusplus\x64\Debug\test_matplotplusplus.exe : fatal error LNK1120: 3 unresolved externals
```

## Remarks

As the errors show, it needs `nodesoup`, which is not in `vcpkg`'s ports.

--- a/ports/matplotplusplus/install-3rd-libraries.patch
+++ b/ports/matplotplusplus/install-3rd-libraries.patch
@@ -1,13 +1,15 @@
 diff --git a/source/3rd_party/CMakeLists.txt b/source/3rd_party/CMakeLists.txt
-index 52f20eb..ab58bbd 100644
+index bef0c08..e7521a7 100644
 --- a/source/3rd_party/CMakeLists.txt
 +++ b/source/3rd_party/CMakeLists.txt
-@@ -41,6 +41,8 @@ endif()
+@@ -41,7 +41,9 @@ endif()
  if(MASTER_PROJECT AND NOT BUILD_SHARED_LIBS)
    install(TARGETS nodesoup
        EXPORT Matplot++Targets
+-      ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}/Matplot++)
 +      RUNTIME DESTINATION bin
 +      LIBRARY DESTINATION lib
-       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}/Matplot++)
++      ARCHIVE DESTINATION lib)
  endif()
+ 
  

--- a/ports/matplotplusplus/portfile.cmake
+++ b/ports/matplotplusplus/portfile.cmake
@@ -49,12 +49,12 @@ vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 
 file(RENAME
-    ${CURRENT_PACKAGES_DIR}/lib/cmake/Matplot++/matplot++-config.cmake
-    ${CURRENT_PACKAGES_DIR}/lib/cmake/Matplot++/Matplot++-config.cmake
+    "${CURRENT_PACKAGES_DIR}/lib/cmake/Matplot++/matplot++-config.cmake"
+    "${CURRENT_PACKAGES_DIR}/lib/cmake/Matplot++/Matplot++-config.cmake"
 )
 
 vcpkg_cmake_config_fixup(PACKAGE_NAME Matplot++ CONFIG_PATH lib/cmake/Matplot++)
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
 
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/matplotplusplus/vcpkg.json
+++ b/ports/matplotplusplus/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "matplotplusplus",
   "version-date": "2021-04-11",
-  "port-version": 2,
+  "port-version": 3,
   "description": "A C++ graphics library for data visualization",
   "homepage": "https://alandefreitas.github.io/matplotplusplus/",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3910,7 +3910,7 @@
     },
     "matplotplusplus": {
       "baseline": "2021-04-11",
-      "port-version": 2
+      "port-version": 3
     },
     "matroska": {
       "baseline": "1.6.2",

--- a/versions/m-/matplotplusplus.json
+++ b/versions/m-/matplotplusplus.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "e60b72bfb17b3263a7ee1e056af221991f5c7d69",
+      "version-date": "2021-04-11",
+      "port-version": 3
+    },
+    {
       "git-tree": "6b6b0deac894d376c04b11f3f01a26952b97b0b1",
       "version-date": "2021-04-11",
       "port-version": 2

